common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -173,12 +174,72 @@ public static String modifyMultipleMatchKeyword(String query){
                         predicates.add(root.get("domainId").in(domainIds));
                     }
 
-                    List<Predicate> countPredicates = new ArrayList<>();
-                    countPredicates.add(criteriaBuilder.greaterThan(root.get("countValue"), 0));
-                    countPredicates.add(criteriaBuilder.greaterThan(root.get("sourceCountValue"), 0));
+                    if(minCount == 1){
+                        List<Predicate> countPredicates = new ArrayList<>();
+                        countPredicates.add(criteriaBuilder.greaterThan(root.get("countValue"), 0));
+                        countPredicates.add(criteriaBuilder.greaterThan(root.get("sourceCountValue"), 0));
+
+                        predicates.add(criteriaBuilder.or(
+                                countPredicates.toArray(new Predicate[0])));
+                    }
+
+                    return criteriaBuilder.and(predicates.toArray(new Predicate[0]));
+                };
+        // Return up to limit results, sorted in descending count value order.
+        Pageable pageable = new PageRequest(0, limit,
+                new Sort(Direction.DESC, "countValue"));
+        NoCountFindAllDao<Concept, Long> conceptDao = new NoCountFindAllDao<>(Concept.class,
+                entityManager);
+        return conceptDao.findAll(conceptSpecification, pageable);
+    }
+
+    public Slice<Concept> searchConcepts(StandardConceptFilter standardConceptFilter, List<String> vocabularyIds, List<String> domainIds, int limit, int minCount) {
```

Comment:
Can't we combine the two searchConcepts methods with some conditional logic based on whether query is null?

Even if you expose two methods like this externally, it'd be nice if their implementation shared code...

Reply:
Removed the other function. Added if's while adding predicates to check if query is null or empty. Thanks Dan.